packages/core/lib/commands/debug.js

```diff
@@ -40,7 +52,12 @@ const command = {
         await Environment.detect(config);
 
         const txHash = config._[0]; //may be undefined
-        return await new CLIDebugger(config).run(txHash);
+        if (config.external && txHash === undefined) {
+          throw new Error(
+            "External mode requires a specific transaction to debug"
+          );
```

Comment:
External mode fetches contracts based on what's called in the specific transaction.  If there's no specific transaction, what's it going to fetch?

Trying to do it once the debugger is already up and running would be quite a problem -- that's why I disallowed loading and unloading in external mode.  There's a reason that light mode to full-mode is a one-time, one-way thing.  I mean notionally you could discard all the `data` state to transition back to light mode, add everything from scratch, then turn on full mode again, but that doesn't fit into our current structure at all, with `CLIDebugger` separate from `DebugCompiler` separate from `DebugInterpreter`, etc.  And the debugger can't *just* clear out the external stuff, leaving the project stuff as a base, because it has no way to distinguish.  Actually, I guess that last part's not quite true, since the compilation IDs would be distinctive, if we're willing to rely on that.  Even so, once it's cleared out, there's not really any way to get it to add the new stuff without breaking structure.  Unless we went with your idea above where the debugger gets an `externalHandler`, but, well, I discussed that below.

I don't know.  This might not be impossible, but I don't really want to try to tackle it right now.  It can perhaps be a later thing?

Reply:
Sounds good, was just curious as to the difficulty!